contrib/mesos/pkg/scheduler/constraint/doc.go

```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package constraint exposes Marathon-like constraints for scheduling pods.
+// Incomplete.
```

Comment:
Initial thinking was that k8s-mesos would support Marathon-like constraints. Perhaps a user could add a pod label such as:
- `k8sm.mesosphere.io/constraint:hostname:UNIQUE`, or
- `k8sm.mesosphere.io/constraint:attribute1:LIKE:value1`

Since the k8s scheduler and k8s-mesos scheduler use different algorithms/rules for pod placement, the k8s scheduler wouldn't have to adopt k8s-mesos scheduling constraints (and vice-versa). I was operating under the assumption that some divergence from k8s would be acceptable here.

Admittedly, more thinking needs to go into this.